### PR TITLE
Document changes to release-process and versioning

### PR DESCRIPTION
**Pull request by arogge**
- docs: update Releasing Bareos
- docs: update Bareos Version Numbers and Releases

### CODE REVIEW

File: docs/manuals/source/DeveloperGuide/ReleasingBareos.rst [hunk 1]
```diff
@@ -19,10 +19,14 @@ So the version you're going to release is probably the final version for that pr
 
 Preparing the release notes
 ---------------------------
-For each new release there should be a section in the :ref:`releasenotes`.
-See the :ref:`associated section <DocumentationStyleGuide/BareosSpecificFormatting/Release:Release Notes>` in the Documentation Style Guide about formatting the release notes.
+In :sinceVersion:`20.0.0: CHANGELOG.md` release notes will be kept in :file:`CHANGELOG.md`.
+The "Unreleased"-section in that file will automatically become the release-notes and a new "Unreleased"-section will be added afterwards.
```
Comment (pstorz) [inline docs/manuals/source/DeveloperGuide/ReleasingBareos.rst:23]:
```suggestion
The "Unreleased" section in that file will automatically become the release notes and a new "Unreleased" section will be added afterwards.
```

File: docs/manuals/source/DeveloperGuide/ReleasingBareos.rst [hunk 2]
```diff
@@ -19,10 +19,14 @@ So the version you're going to release is probably the final version for that pr
...
 
-The release notes should be committed to the master branch and then cherry-picked to your release branch using ``git cherry-pick -x``.
+How does it work in detail?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+When you run :file:`devtools/prepare-release.sh` the "Unreleased"-section will be renamed to the current release by :file:`devtools/new-changelog-release.sh`.
```
Comment (pstorz) [inline docs/manuals/source/DeveloperGuide/ReleasingBareos.rst:27]:
```suggestion
When you run :file:`devtools/prepare-release.sh` the "Unreleased" section will be renamed to the current release by :file:`devtools/new-changelog-release.sh`.
```

File: docs/manuals/source/DeveloperGuide/ReleasingBareos.rst [hunk 3]
```diff
@@ -19,10 +19,14 @@ So the version you're going to release is probably the final version for that pr
...
+All links to pull-requests, issues, etc. will be automatically updated by :file:`devtools/update-changelog-links.sh`.
```
Comment (pstorz) [inline docs/manuals/source/DeveloperGuide/ReleasingBareos.rst:28]:
```suggestion
All links to pull requests, issues, etc. will be automatically updated by :file:`devtools/update-changelog-links.sh`.
```

File: docs/manuals/source/DeveloperGuide/ReleasingBareos.rst [hunk 4]
```diff
@@ -19,10 +19,14 @@ So the version you're going to release is probably the final version for that pr
...
+After the release has been made :file:`devtools/new-changelog-release.sh` will be called again to add the new "Unreleased"-section that will be added to the commit for the work-in-progress tag.
```
Comment (pstorz) [inline docs/manuals/source/DeveloperGuide/ReleasingBareos.rst:29]:
```suggestion
After the release has been made :file:`devtools/new-changelog-release.sh` will be called again to add the new "Unreleased" section that will be added to the commit for the work-in-progress tag.
```

**Review by pstorz — APPROVED** [review]
I have just some minor suggestions to remove in my view unneded dashes (-).

**Review by franku — APPROVED** [review]
See separate commit for suggestions

**Pull request merged**